sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/storage/fileshare/AzureStorageFileShareAutoConfigurationTest.java

```diff
@@ -97,6 +101,102 @@ void shareNameNotSetShouldNotConfigureShareClient() {
+        ShareClient shareClient = mock(ShareClient.class);
+        when(shareServiceClient.getShareClient("share1")).thenReturn(shareClient);
+        when(shareClient.getFileClient("/directory1/file1")).thenReturn(mock(ShareFileClient.class));
+
+        ShareServiceAsyncClient shareServiceAsyncClient = mock(ShareServiceAsyncClient.class);
+        ShareAsyncClient shareAsyncClient = mock(ShareAsyncClient.class);
+        when(shareServiceAsyncClient.getShareAsyncClient("share1")).thenReturn(shareAsyncClient);
+        when(shareAsyncClient.getFileClient("/directory1/file1")).thenReturn(mock(ShareFileAsyncClient.class));
+        this.contextRunner
+            .withPropertyValues(
+                "spring.cloud.azure.storage.fileshare.account-name=sa",
+                "spring.cloud.azure.storage.fileshare.share-name=share1",
+                "spring.cloud.azure.storage.fileshare.file-path=/directory1/file1"
+            )
+            .withBean(AzureGlobalProperties.class, AzureGlobalProperties::new)
+            .withBean(ShareServiceClient.class, () -> shareServiceClient)
+            .withBean(ShareServiceAsyncClient.class, () -> shareServiceAsyncClient)
+            .run(context -> {
+                assertThat(context).hasSingleBean(ShareFileClient.class);
+                assertThat(context).hasSingleBean(ShareFileAsyncClient.class);
+            });
+    }
+
+    @Test
+    void filePathNotSetShouldNotConfigureFileClient() {
+        ShareServiceClient shareServiceClient = mock(ShareServiceClient.class);
+        when(shareServiceClient.getShareClient("share1")).thenReturn(mock(ShareClient.class));
+
+        ShareServiceAsyncClient shareServiceAsyncClient = mock(ShareServiceAsyncClient.class);
+        when(shareServiceAsyncClient.getShareAsyncClient("share1")).thenReturn(mock(ShareAsyncClient.class));
+
+        this.contextRunner
+            .withPropertyValues(
+                "spring.cloud.azure.storage.fileshare.account-name=sa",
+                "spring.cloud.azure.storage.fileshare.share-name=share1"
+            )
+            .withBean(AzureGlobalProperties.class, AzureGlobalProperties::new)
+            .withBean(ShareServiceClient.class, () -> mock(ShareServiceClient.class))
+            .withBean(ShareServiceAsyncClient.class, () -> mock(ShareServiceAsyncClient.class))
+            .run(context -> {
+                assertThat(context).doesNotHaveBean(ShareFileClient.class);
+                assertThat(context).doesNotHaveBean(ShareFileAsyncClient.class);
+            });
+    }
+
+    @Test
+    void directoryPathSetShouldConfigureDirectoryClient() {
+        ShareServiceClient shareServiceClient = mock(ShareServiceClient.class);
+        ShareClient shareClient = mock(ShareClient.class);
+        when(shareServiceClient.getShareClient("share1")).thenReturn(shareClient);
+        when(shareClient.getDirectoryClient("/directory1/directory2")).thenReturn(mock(ShareDirectoryClient.class));
```

Comment:
I think the file-path and directory-path doesn't have `/` at the beginning.

/directory1/directory2 --> `directory1/directory2`

![image](https://user-images.githubusercontent.com/4465723/150727696-f8ee0f14-e5bc-4452-a096-e96922fe5342.png)